ostia-operator/test/tests/01-simple_endpoint/run.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+set -e
+
+source ../common/common.sh
+
+ns="simple-api-$(generate_random_string)"
+endpoint="${ns}.${OPENSHIFT_PUBLIC_HOSTNAME}"
+
+echo "Testing Simple API Endpoint in namespace ${ns}"
+
+oc new-project ${ns} > /dev/null
+
+../common/deploy_operator.sh ${ns}
+
+oc new-app -f ../common/echo_api.yaml -n ${ns} >/dev/null
+
+wait_for_pod_ready app echo-api ${ns}
+
+echo "echo-api deployed in ${ns}"
+
+oc new-app -f ./endpoint.yaml --param HOSTNAME=${endpoint} -n ${ns} >/dev/null
+
+wait_for_pod_ready app apicast ${ns}
+
+echo "Proxy deployed successfully in ${ns}"
+
+# Verifying expected HTTPS status
+
+result=$(do_http_get ${endpoint}"/hello" 10)
+
+# Cleanup
+oc delete project ${ns}
```

Comment:
clean up is also ripe for extracting out of the specific tests, ie. we could specify what each tests needs as parameters to some setup function:
```bash
setup_test setup_fn, cleanup_fn, test_fn
```
where each function would correspond to each part of the test, and other helpers would be available to provide quicker default convenience functions, `setup_ns_with_apicast test_fn`. Most of the time just writing the `test_fn` would be the only needed step.

Reply:
As per conversation on Slack, cleanup will be moved into its own `make` target and namespaces labelled and deleted before running the test suite. Anything you see missing from this?

Reply:
not at this point, no :+1: